### PR TITLE
[GEN-226] Support absolute URLs for the 'next' query param

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Support passing absolute URLs in the "next" query parameter.

## Justify why they are needed

This will be used by AdTraction for affiliate links.

We will still not allow any next redirects to go to a different domain than the current one.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2425** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2425" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2425?utm_source=stack-comment).

File: apps/store/src/utils/getPathnameFromUrl.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const getPathnameFromUrl = (rawUrl: string) => {
+  try {
+    return new URL(rawUrl).pathname
```
Comment (alebedev) [inline apps/store/src/utils/getPathnameFromUrl.ts:3]: Nit: you can avoid error when `rawUrl` is path-only:
```
new URL(rawUrl, ORIGIN_URL)
```

**Review by alebedev — APPROVED** [review]
Nice, one comment inside

**Review by robinandeer — COMMENTED** [review]
Hmm you mean we should check first if the URL is relative or absolute?

File: apps/store/src/utils/getPathnameFromUrl.ts (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/utils/getPathnameFromUrl.ts:3]: Hmm you mean we should check first if the URL is relative or absolute?

**Pull request merged**